libraries/helpers.rb

```diff
@@ -0,0 +1,60 @@
+module OslRepos
+  module Cookbook
+    module Helpers
+      # Define variables to use in repo urls
+      def centos_url
+        # CentOS 7 splits up ppc64 and aarch64 into a secondary architecture repo.
+        if node['platform_version'].to_i < 8
+          case node['kernel']['machine']
+          when 'aarch64', 'ppc64le', 's390x'
+            url = 'https://centos-altarch.osuosl.org'
+          else
+            url = 'https://centos.osuosl.org'
+          end
+        else
+          url = 'https://centos.osuosl.org'
+        end
+        url
+      end
+
+      # power8/9 replaces $basearch in power9 repositories
+      def base_arch
+        if power8?
+          'power8'
+        elsif power9?
+          'power9'
+        else
+          '$basearch'
+        end
+      end
+
+      # Is this a power8 architecture?
+      def power8?
+        if node['ibm_power'] && node['ibm_power']['cpu']
+          if node['kernel']['machine'] == 'ppc64le' || !node['ibm_power'].nil?
+            node['ibm_power']['cpu']['cpu_model'] =~ /power8/
+          else
+            false
+          end
+        else
+          false
+        end
+      end
+
+      # Is this a power8 architecture?
+      def power9?
+        if node['ibm_power'] && node['ibm_power']['cpu']
+          if node['kernel']['machine'] == 'ppc64le' || !node['ibm_power'].nil?
```

Comment:
I believe this is caught above and isn't needed. Same for `power8?`.
```suggestion
          if node['kernel']['machine'] == 'ppc64le'
```